oxide-auth/src/primitives/generator.rs

```diff
@@ -160,14 +160,21 @@ impl Assertion {
     fn extract<'a>(&self, token: &'a str) -> Result<(Grant, String), ()> {
         let decoded = decode(token).map_err(|_| ())?;
         let assertion: AssertGrant = rmp_serde::from_slice(&decoded).map_err(|_| ())?;
-        ring::hmac::verify_with_own_key(&self.secret, &assertion.0, &assertion.1).map_err(|_| ())?;
+
+        let mut hmac = Hmac::<sha2::Sha256>::new_varkey(&self.key).unwrap();
```

Comment:
Maybe this performs the key-transformation? If so, store this struct and not the raw key. If it is possible to clone it here, that is.

Reply:
That might be the right thing to do, but I figured that since you used `SecretKey` and that wrapped the intent to use SHA256 at construction time and I couldn't figure out how to get a similar result, that having the key available for reuse might be of some use. It does feel wrong tho. That's what I alluded to in my 1st comment.